tests/s3/test_bucket_acl.py

```diff
@@ -46,58 +46,62 @@
 class TestBucketACL:
     """Bucket ACL Test suite."""
 
-    @classmethod
-    def setup_class(cls):
-        """Function will be invoked prior to each test case."""
-        cls.log = logging.getLogger(__name__)
-        cls.log.info("STARTED: setup test operations.")
-        cls.s3_obj = s3_test_lib.S3TestLib(endpoint_url=S3_CFG["s3_url"])
-        cls.iam_obj = iam_test_lib.IamTestLib(endpoint_url=S3_CFG["iam_url"])
-        cls.acl_obj = s3_acl_test_lib.S3AclTestLib(endpoint_url=S3_CFG["s3_url"])
-        cls.test_file = f"testfile{time.perf_counter()}.txt"
-        cls.s3_passwd = S3_CFG["CliConfig"]["s3_account"]["password"]
-        cls.test_dir_path = os.path.join(TEST_DATA_PATH, "TestBucketACL")
-        if not system_utils.path_exists(cls.test_dir_path):
-            resp = system_utils.make_dirs(cls.test_dir_path)
-            cls.log.info("Created path: %s", resp)
-        cls.log.info("Test data path: %s", cls.test_dir_path)
-        cls.test_file_path = os.path.join(cls.test_dir_path, cls.test_file)
-        cls.bucket = "{}-{}".format("aclbucket", time.perf_counter_ns())
-        cls.account_prefix = "acltestaccn_{}"
-        cls.account = "{}{}".format("acltestaccn1", time.perf_counter_ns())
-        cls.email_id = "{}{}".format(cls.account, "@seagate.com")
-        cls.account1 = "{}{}".format("acltestaccn2", time.perf_counter_ns())
-        cls.email_id1 = "{}{}".format(cls.account, "@seagate.com")
-        cls.rest_obj = S3AccountOperations()
-        cls.log.info("Bucket name: %s", cls.bucket)
-        cls.account_list = []
-        cls.log.info("ENDED: Setup test operations")
-
-    @classmethod
-    def teardown_class(cls):
-        """Function to perform the cleanup for test suit."""
-        cls.log.info("STARTED: Teardown test operations")
-        if system_utils.path_exists(cls.test_file_path):
-            resp = system_utils.remove_file(cls.test_file_path)
-            cls.log.info("removed path: %s, resp: %s", cls.test_file_path, resp)
-        cls.log.info("Deleting buckets in default account")
-        resp = cls.s3_obj.bucket_list()
-        cls.log.info(resp)
-        pref_list = [each_bucket for each_bucket in resp[1] if each_bucket == cls.bucket]
+    @pytest.fixture(autouse=True)
```

Comment:
You can add # pylint: disable=attribute-defined-outside-init to skip the codacy attribute defined outside inti issue.